src/cljc/athens/common_events/resolver/undo.cljc

```diff
@@ -66,6 +66,20 @@
     [(atomic-graph-ops/make-block-remove-op uid)]))
 
 
+(defmethod resolve-atomic-op-to-undo-ops :page/remove
+  [_db evt-db {:op/keys [args]}]
+  (let [{:page/keys [title]} args
+        {page-refs :block/_refs} (common-db/get-page-document evt-db [:node/title title])
+        page-repr                 [(common-db/get-internal-representation evt-db (:db/id (d/entity evt-db [:node/title title])))]
+        repr-ops                  (bfs/internal-representation->atomic-ops evt-db page-repr nil)
+        save-ops                  (->> page-refs
+                                       (map :db/id)
+                                       (map (partial common-db/get-block evt-db))
+                                       (map (fn [{:block/keys [uid string]}]
+                                              (atomic-graph-ops/make-block-save-op uid string))))]
```

Comment:
Can you leave a comment here about how it also needs to restore the shortcut? I'll add it later myself, when I wrap up Jeffs changes.